source/renderer/app/containers/LoadingPage.js

```diff
@@ -47,23 +64,54 @@ export default class LoadingPage extends Component<InjectedProps> {
           hasLoadedCurrentLocale={hasLoadedCurrentLocale}
           hasLoadedCurrentTheme={hasLoadedCurrentTheme}
           currentLocale={currentLocale}
-          handleReportIssue={this.handleReportIssue}
-          onProblemSolutionClick={this.handleProblemSolutionClick}
+          onExternalLinkClick={stores.app.openExternalLink}
+          onReportIssueClick={this.handleReportIssueClick}
           onCheckTheTimeAgain={forceCheckLocalTimeDifference}
           onContinueWithoutClockSyncCheck={ignoreSystemTimeChecks}
+          onDownloadLogs={this.handleDownloadLogs}
         />
-        <WalletSupportRequestPage />
+        <NotificationMessage
+          icon={successIcon}
+          show={stores.uiNotifications.isOpen(id)}
+          onClose={() => this.props.actions.notifications.closeActiveNotification.trigger({ id })}
+          clickToClose
+          hasCloseButton
+        >
+          {message}
+        </NotificationMessage>
       </CenteredLayout>
     );
   }
 
-  handleReportIssue = () => {
-    this.props.actions.dialogs.open.trigger({
-      dialog: BugReportDialog
-    });
+  get notification() {
+    const { intl } = this.context;
+    return {
+      id: 'loading-page-download-logs-success',
+      duration: DOWNLOAD_LOGS_SUCCESS_DURATION,
+      message: intl.formatMessage(messages.downloadLogsSuccess),
+    };
+  }
+
+  handleReportIssueClick = async (event: SyntheticEvent<HTMLButtonElement>) => {
+    event.persist();
+    const { intl } = this.context;
+    const reportIssueButtonUrl = intl.formatMessage(messages.reportIssueButtonUrl);
+    const locale = this.props.stores.profile.currentLocale;
+    const supportUrl = await getSupportUrl(reportIssueButtonUrl, locale);
+    this.props.stores.app.openExternalLink(supportUrl);
   };
 
-  handleProblemSolutionClick = (url: string) => {
-    openExternalUrlChannel.send(url);
+  handleDownloadLogs = () => {
+    const { actions } = this.props;
+    const fileName = generateFileNameWithTimestamp();
+    const destination = global.dialog.showSaveDialog({
+      defaultPath: fileName,
+    });
+    if (destination) {
+      actions.profile.downloadLogs.trigger({ fileName, destination, fresh: true });
+    }
+    const { id, duration } = this.notification;
+    actions.notifications.open.trigger({ id, duration, });
```

Comment:
You forgot to check if the downloaded logs are *actually* successfully saved before showing the notification!

Reply:
You are right! Do you think listening to the action is enough or we should actually check if the file exists?